okhttp-sse/src/main/java/okhttp3/internal/sse/ServerSentEventReader.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal.sse
+
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString.Companion.encodeUtf8
+import okio.Options
+import java.io.IOException
+
+class ServerSentEventReader(
+  private val source: BufferedSource,
+  private val callback: Callback
+) {
+  private var lastId: String? = null
+
+  interface Callback {
+    fun onEvent(id: String?, type: String?, data: String)
+    fun onRetryChange(timeMs: Long)
+  }
+
+  /**
+   * Process the next event. This will result in a single call to [Callback.onEvent] *unless* the
+   * data section was empty. Any number of calls to [Callback.onRetryChange] may occur while
+   * processing an event.
+   *
+   * @return false when EOF is reached
+   */
+  @Throws(IOException::class)
+  fun processNextEvent(): Boolean {
+    var id = lastId
+    var type: String? = null
+    val data = Buffer()
+
+    while (true) {
+      val option = source.select(options)
+      when (option) {
+        0, 1, 2 -> {
```

Comment:
gross . . . ideas?

Reply:
`0..2`?

I thought there was a reason that I avoided using select... I don't recall it.

Reply:
Fixed.